microhapulator/filter.py

```diff
@@ -106,13 +125,6 @@ def keep(self, read):
         is_ambiguous = read.seq.count("N") / len(read.seq) > self.threshold
         return not is_ambiguous
 
-    def write_counts_output(self):
-        with self.outfiles["counts"] as counts_out:
-            counts_out.write(
-                f"ReadsRemoved\tReadsKept\n{self.num_reads_failed}\t{self.num_reads_passed}\n"
-            )
-
-    def get_output_files(self, out_prefix):
-        self.outfiles = dict()
-        self.outfiles["reads"] = mhopen(f"{out_prefix}-ambig-filtered.fastq", "w")
-        self.outfiles["counts"] = mhopen(f"{out_prefix}-ambig-read-counts.txt", "w")
+    @property
+    def summary(self):
+        return f"ReadsRemoved\tReadsKept\n{self.num_reads_failed}\t{self.num_reads_passed}"
```

Comment:
I refactored the filter classes to return data rather than write to a file. I had originally refactored them to return the various values themselves, leaving the formatting to external code. But I walked that back a bit. So I'm open to negotiation here. 😎